rust/kernel/endian.rs

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Endian integer types
+
+macro_rules! define_le_integer {
+    ($name:ident, $native_type:ty) => {
+        /// little-endian integer
```

Comment:
```suggestion
        /// Little-endian integer.
```

Ideally, a bit more documentation would be nice (you may use the `doc` attribute to construct documentation in the macro as needed using `$name` etc.).

I see you used the standard library examples for the tests below (thanks for writing tests!), so it would be nice to have them in the docs in a small `# Examples` section, like the standard library.

Reply:
Thanks, improved the doc with `doc` attribute (added Examples session too).